src/iot_logging.h

```diff
@@ -0,0 +1,54 @@
+#include <stdio.h>
+#define STRINGIFY(a) (#a)
+
+/*
+ESP8266 has limited RAM so we force all logging strings to PROGMEM (flash)
+and we leverage os_printf rather than printf.
+
+Time inclusion in errors is possible if NTP was successful, and can be added
+in a later revision.
+*/
+
+#if defined(ARDUINO_ARCH_ESP8266)
+#include "esp8266/azcpgmspace.h"
+#define LogUsage(FORMAT, ...) { \
```

Comment:
Yeah, the end goal is to have as much in the upstream repo as possible. However there are still a few things to figure out to ensure smooth updates to this repo and making sure it ifs in with all the other language sdks in the upstream. I'll we can open an issue on the upstream repo if one doesn't exist for this, but i don't think we need to wait for that before merging this PR.